pubtools/_pulp/tasks/copy_repo.py

```diff
@@ -0,0 +1,200 @@
+import logging
+from functools import partial
+
+import attr
+from more_executors.futures import f_map, f_sequence
+from pubtools.pulplib import (
+    ContainerImageRepository,
+    Criteria,
+    ErratumUnit,
+    FileUnit,
+    ModulemdUnit,
+    RpmUnit,
+)
+
+from pubtools._pulp.arguments import SplitAndExtend
+from pubtools._pulp.services import CollectorService, PulpClientService
+from pubtools._pulp.task import PulpTask
+from pubtools._pulp.tasks.common import PulpRepositoryOperation
+
+step = PulpTask.step
+
+LOG = logging.getLogger("pubtools.pulp")
+
+
+@attr.s(slots=True)
+class RepoCopy(object):
+    """Represents a copy of a single repo."""
+
+    tasks = attr.ib()
+    """The completed Pulp tasks for copying."""
+
+    repo = attr.ib()
+    """The repo to which content was copied."""
+
+
+class CopyRepo(CollectorService, PulpClientService, PulpRepositoryOperation):
+    @property
+    def content_type(self):
+        # Only return non-None if there were really any types given.
+        # Otherwise, return None to let library defaults apply
+        c = self.args.content_type or None
+        # Normalize content types (e.g., "ISO" -> "iso").
+        if c:
+            c = [t.lower() for t in c]
+        return c
+
+    @property
+    def repo_pairs(self):
+        return [pair.split(",") for pair in self.args.repopairs]
```

Comment:
I think either here, or in `get_repos`, we ought to be validating that we really have pairs.

I'm reading through how this is used in `get_repos` and it seems pretty easy for the caller to mess up and have the command not do what was intended.  For example if the user accidentally types:

```
copy-repo ... repo1,repo2,repo3,repo4
```

...where they meant to type `repo1,repo2 repo3,repo4`, it won't detect a problem.  It seems like it'll just do the copy between repo1,repo2 and ignore repo3,repo4 with no indication to the user that something went wrong.